### PR TITLE
Fix architect.Construct to avoid throwing incorrect RangeErrors

### PR DESCRIPTION
**Pull request by AkashSamlal**
Fix issue #78

### CODE REVIEW

File: test/units/architecture/architect.test.js [hunk 1]
```diff
@@ -1,22 +1,34 @@
 let { assert, expect } = require('chai');
 let should = require('chai').should();
-let carrot = require('../../../src/carrot');
+let {architect, Network, Node } = require('../../../src/carrot');
```
Comment (luiscarbonell) [inline test/units/architecture/architect.test.js]: We can start turning all of these imports in to `const`s they should not be overwritten throughout testing and we can save on the memory.

File: test/units/architecture/architect.test.js [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
  */
-it('should return true if valid user id', function(){
+describe('architect', function(){
+  describe('Perceptron', function() {
+    it('given input: 0, hidden: 0, output: 0, should throw Error', function(){
+           assert.throws(function() {  return new architect.Perceptron(0, 0, 0)  }, RangeError);
+       });
   
-});
+       it('should return false if invalid user id', function(){
```
Comment (luiscarbonell) [inline test/units/architecture/architect.test.js]: What is _"user id"_? 

Was this ported from StackOverflow? 

Did we add a `user_id` to `Network`?

Comment (AkashSamlal) [reply test/units/architecture/architect.test.js]: No, the first two tests was a walkthrough with Chris and I

Comment (luiscarbonell) [reply test/units/architecture/architect.test.js]: Got you! 

Let's not have those bleed through to production code.

**Comment by luiscarbonell** [comment]
@AkashSamlal Going to close for now my man. 

If you can pull again when you made the changes we just talked about, that would be 👌

File: src/architecture/architect.js [hunk 3]
```diff
@@ -5,562 +5,596 @@ var Layer = require('./layer');
...
+*
+* // Input: 2 neurons, 4 Hidden layers: 10 neurons, Output: 1 neuron
+* let myPerceptron = new architect.Perceptron(2, 10, 10, 10, 10, 1);
+*
+* @returns {Network} Feed forward neural network
+*/
+Perceptron: function () {
+  // Convert arguments to Array
+  // Coerce arguments to be Array-like w/ .call
+  var sizes = Array.prototype.slice.call(arguments); //type-casting/coercion arguments into array
```
Comment (dan-ryan) [inline src/architecture/architect.js]: You can use the newer `Array.from` instead these days.

Comment (dan-ryan) [reply src/architecture/architect.js]: https://stackoverflow.com/questions/36995617/convert-array-like-objects-array-prototype-slice-or-array-from

**Pull request closed without merging**